sec_certs/dataset/common_criteria.py

```diff
@@ -684,12 +684,17 @@ def _compute_normalized_cert_ids(self):
         for cert in self:
             cert.compute_heuristics_cert_id(self.all_cert_ids)
 
+    def _compute_dependency_vulnerabilities(self):
+        for cert in self:
+            cert.find_certificate_dependency_vulnerabilities(self)
```

Comment:
I think that we have unified interface for vulnerabilities and dependencies between CC and FIPS. Would it be possible to generalize this method and implement it into abstract `Dataset` class, so that it works both on FIPS and CC? While irrelevant for your thesis, I think that it's little extra work with nice benefit. 

Just one caveat: I think that we're using different objects for holding `cert_id` in the respective frameworks. Also, FIPS cert_ids are 100% accurate so no need to check for duplicites, etc.